python/lsst/ip/isr/isrMock.py

```diff
@@ -0,0 +1,915 @@
+        """
+        for x in range(0, ampData.getDimensions().getX()):
+            for y in range(0, ampData.getDimensions().getY()):
+                ampData.getArray()[y][x] = (ampData.getArray()[y][x] +
+                                            scale * np.exp(-0.5 * ((x - 50)**2 + (y - 25)**2)**2 / 3.0**2))
+
+    def amplifierAddCT(self, ampDataSource, ampDataTarget, scale):
+        r"""Add a scaled copy of an amplifier to another, simulating crosstalk.
+
+         This method operates in the amplifier coordinate frame.
+
+        Parameters
+        ----------
+        ampData : `lsst.afw.image.ImageF`
+            Amplifier image to operate on.
+        scale : float
+            Peak flux of the source to add.
+
+        Notes
+        -----
+        This simulates simple crosstalk between amplifiers.
+        """
+        for x in range(0, ampDataSource.getDimensions().getX()):
+            for y in range(0, ampDataSource.getDimensions().getY()):
+                ampDataTarget.getArray()[y][x] = (ampDataTarget.getArray()[y][x] +
+                                                  scale * ampDataSource.getArray()[y][x])
+
+    # Functional form data values
+    def amplifierAddFringe(self, amp, ampData, scale):
+        r"""Add a fringe-like ripple pattern to an amplifier's image data.
+
+        Parameters
+        ----------
+        amp : `lsst.afw.ampInfo.AmpInfoRecord`
+            Amplifier to operate on.  Needed for amp<->exp coordinate transforms.
+        ampData : `lsst.afw.image.ImageF`
+            Amplifier image to operate on.
+        scale : float
+            Peak intensity scaling for the ripple.
+
+        Notes
+        -----
+        This uses an offset sinc function to generate a ripple
+        pattern.  True fringes have much finer structure, but this
+        pattern should be visually identifiable.  The (x, y)
+        coordinates are in the frame of the amplifier, and (u, v) in
+        the frame of the full trimmed image.
+        """
+        for x in range(0, ampData.getDimensions().getX()):
+            for y in range(0, ampData.getDimensions().getY()):
+                (u, v) = self.localCoordToExpCoord(amp, x, y)
+                ampData.getArray()[y][x] = (ampData.getArray()[y][x] +
+                                            scale * np.sinc(((u - 100)/50)**2 + (v / 50)**2))
+
+    def amplifierAddFlat(self, amp, ampData, size):
+        r"""Add a flat-like dome pattern to an amplifier's image data.
+
+        Parameters
+        ----------
+        amp : `lsst.afw.ampInfo.AmpInfoRecord`
+            Amplifier to operate on.  Needed for amp<->exp coordinate transforms.
+        ampData : `lsst.afw.image.ImageF`
+            Amplifier image to operate on.
+        size : float
+            Size of the dome to use.  A larger value results in a
+            smaller center-to-edge fall off.
+
+        Notes
+        -----
+        This uses a centered 2-d Gaussian to simulate an illumination
```

Comment:
I'm not sure how this is "centered" as you say. It seems like it would be 100 pixels from each edge, but there's no guarantee that that's the center.

Reply:
For this afw.cameraGeom detector, 100 pixels does center the Gaussian.  As part of making the scale factor more intuitive, I've added the actual center as a parameter that can be set.